pkg/awsutils/awsutils.go

```diff
@@ -185,6 +185,9 @@ type APIs interface {
 
 	// GetInstanceID returns the instance ID
 	GetInstanceID() string
+
+	// Verify if the InstanceNetworkingLimits has the ENI limits else make EC2 call to fill cache.
+	VerifyElseFetchLimitsFromEC2() error
```

Comment:
this name feels a bit odd..
i'm not a naming expert either, but how about `FetchInstanceTypeLimits` or `EnsureInstanceTypeLimits` 😄

Reply:
Yeah sure :)